skl/arena.go

```diff
@@ -35,11 +36,22 @@ const (
 
 // Arena should be lock-free.
 type Arena struct {
-	*z.Buffer
+	data    []byte
+	offset  uint32
+	Release func()
 }
 
 func (s *Arena) size() int64 {
-	return int64(s.LenNoPadding())
+	return int64(atomic.LoadUint32(&s.offset))
+}
+
+func (s *Arena) allocate(sz uint32) uint32 {
+	offset := atomic.AddUint32(&s.offset, sz)
+	if offset >= uint32(len(s.data)) {
+		log.Fatalf("Arena too small, toWrite:%d newTotal:%d limit:%d",
+			sz, offset+sz, len(s.data))
```

Comment:
new total should be just `offset` since you've already added it on line 49.

Reply:
Done